plugin/cypress/support/commands.ts

```diff
@@ -25,13 +25,29 @@
 // -- This will overwrite an existing command --
 // Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
 //
-// declare global {
-//   namespace Cypress {
-//     interface Chainable {
-//       login(email: string, password: string): Chainable<void>
-//       drag(subject: string, options?: Partial<TypeOptions>): Chainable<Element>
-//       dismiss(subject: string, options?: Partial<TypeOptions>): Chainable<Element>
-//       visit(originalFn: CommandOriginalFn, url: string, options: Partial<VisitOptions>): Chainable<Element>
-//     }
-//   }
-// }
\ No newline at end of file
+declare global {
+    namespace Cypress {
+        interface Chainable {
+            login(OC_CLUSTER_USER: string, OC_CLUSTER_PASS: string, OC_IDP: string): Chainable<void>
+            //   drag(subject: string, options?: Partial<TypeOptions>): Chainable<Element>
+            //   dismiss(subject: string, options?: Partial<TypeOptions>): Chainable<Element>
+            //   visit(originalFn: CommandOriginalFn, url: string, options: Partial<VisitOptions>): Chainable<Element>
+        }
+    }
+}
+
+Cypress.Commands.add('login', (OC_CLUSTER_USER, OC_CLUSTER_PASS, OC_IDP) => {
+    const user = OC_CLUSTER_USER || Cypress.env('OC_CLUSTER_USER')
+    const password = OC_CLUSTER_PASS || Cypress.env('OC_CLUSTER_PASS')
+    const idp = OC_IDP || Cypress.env('OC_IDP')
+
+    cy.visit('/').then(() => {
+        cy.log('OC_IDP: ', typeof (idp), JSON.stringify(idp))
+        if (idp != undefined) {
+            cy.get('.pf-c-button').contains(idp).click()
+        }
+        cy.get('#inputUsername').type(user)
+        cy.get('#inputPassword').type(password)
+        cy.get('button[type="submit"]').click();
+    })
+})
```

Comment:
What this does is login to openshift console, save the cookie with `cy.session`, then the console plugin uses that login cookie to authenticate with the kiali api? I think this is ok for the console plugin since I think the only auth strategy the console plugin will work with is openshift? Whereas standalone Kiali must with work a number of different auth strategies.

Reply:
That is a great question. I'm not aware if there are any restrictions regarding authorization for OCP to display OSSMC. For this case I followed the simplest scenario to authorize into the console and smoke test the integration which is essential for the release pipeline.

Reply:
The `cy.session` usage is based on this [approach from the docs](https://docs.cypress.io/api/commands/session#Caching-session-data-across-specs) . All is done in the UI, there is room for improvements, but the most important thing is that the login routine is not triggered each scenario or spec file and it's cached for reuse. Additionally I want to implement session validation (see in the link above) so tests could be even faster (nice to have for now).